exonum/src/runtime/mod.rs

```diff
@@ -359,10 +373,34 @@ impl<'a> ExecutionContext<'a> {
     pub(crate) fn dispatch_action(&mut self, action: dispatcher::Action) {
         self.actions.push(action);
     }
+}
 
-    pub(crate) fn take_actions(&mut self) -> Vec<dispatcher::Action> {
-        let mut other = Vec::new();
-        std::mem::swap(&mut self.actions, &mut other);
-        other
+/// Instance descriptor contains information to access running service instance.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct InstanceDescriptor<'a> {
```

Comment:
I'd suggest expanding the docs on id/name slightly: 

https://github.com/dmitry-timofeev/exonum-java-binding-1/blob/ddadb22fd66a52b51d0541fcb99aee67dce65b62/exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceInstanceSpec.java#L12-L25